strings/knuth_morris_prat.dart

```diff
@@ -0,0 +1,95 @@
+import 'package:test/test.dart';
+
+/// Preprocess pattern to identify any suffixes that are identical to prefixes
+/// in the given string.
+///
+/// Build a pattern which tells us where to continue iteration from if we
+///  get a mismatch between the character
+///
+/// Step through the text one character at a time and compare it to a character in
+/// the pattern updating our location within the pattern if necessary
+bool stringCompare(String string, String subString) {
+  if (subString.isEmpty || string.isEmpty) {
+    return true;
+  }
```

Comment:
I don't know if it's good to return `True` if we are searching something in empty string.
```dart
  test(('Searching in empty string'), () {
    string = '';
    subString = 'asd';
    expect(stringCompare(string, subString), isFalse);
  });
```

Reply:
i think it is fine, because this is the way of KMP algorithm. For a empty string there is not pattern in this empty string which is similar to subtring `asd` thus returing false is fine.

Reply:
You mean that test is correct? Because right now it's failing, when `string.isEmpty` `true` is always returned, even when `subString` is not empty.

Reply:
Yup i mean that this test is correct, if this test is failing it means the implementaion of KMP algorithm is not correct, it is buggy. returned result should be `false` in this case.